### PR TITLE
Fix #8614 that hybrid unitary HVAC equipment cannot be found

### LINKED ISSUE #8614: Hybrid Unitary HVAC equipment cannot be found
Issue overview
--------------
The name for the object ZoneHVAC: HybridUnitaryHVAC can only be correctly recognized if it is written in capital letter from users' inputs. If small letters are part of the name for HybridUnitaryHVAC , the simulation will run into the fatal error as follows. "SimZoneHybridUnitaryAirConditioners: Zone evaporative cooler unit not found."

### Details
Some additional details for this issue (if relevant):
 - EnergyPlus v9.5  
 - helpdesk ticket number #15653

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added (list location of defect file here)
 - [x] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by LipingWang**
Pull request overview
---------------------
 - Fixes #8614 
 - Update Energy+.idd.in by removing "retaincase" property for the field name of ZoneHVAC:HybridUnitaryHVAC.  
   
### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: idd/Energy+.idd.in [hunk 1]
```diff
@@ -35433,7 +35433,6 @@ ZoneHVAC:HybridUnitaryHVAC,
     A1, \field Name
         \required-field
         \type alpha
-        \retaincase
```
Comment (Myoldmopar) [inline idd/Energy+.idd.in:35436]: Yeah I think this is good, I don't know why it was retaincase before.  There shouldn't be anything case specific.

File: src/EnergyPlus/HybridUnitaryAirConditioners.cc [hunk 2]
```diff
@@ -99,7 +99,7 @@ void SimZoneHybridUnitaryAirConditioners(EnergyPlusData &state,
     if (CompIndex == 0) {
         CompNum = UtilityRoutines::FindItemInList(CompName, state.dataHybridUnitaryAC->ZoneHybridUnitaryAirConditioner);
         if (CompNum == 0) {
-            ShowFatalError(state, "SimZoneHybridUnitaryAirConditioners: Zone evaporative cooler unit not found.");
+            ShowFatalError(state, "SimZoneHybridUnitaryAirConditioners: ZoneHVAC:HybridUnitaryHVAC not found.");
```
Comment (Myoldmopar) [inline src/EnergyPlus/HybridUnitaryAirConditioners.cc:102]: Yeah that's better. 👍

File: tst/EnergyPlus/unit/UnitaryHybridAirConditioner.unit.cc [hunk 3]
```diff
@@ -515,18 +515,18 @@ TEST_F(EnergyPlusFixture, Test_UnitaryHybridAirConditioner_ValidateFieldsParsing
 {
     std::string idf_objects = delimited_string({
         "ZoneHVAC:HybridUnitaryHVAC,",
-        "MUNTERSEPX5000,          !- Name",
+        "Hybrid Unit 1,          !- Name",
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/UnitaryHybridAirConditioner.unit.cc:518]: I'm not sure if this change is actually tested in the unit test, is it?  I'm OK with this simple IDD change anyway.  @mjwitte is there something I'm missing that this shouldn't go right in?

**Review by Myoldmopar — APPROVED** [review]

File: tst/EnergyPlus/unit/UnitaryHybridAirConditioner.unit.cc (hunk 3 above)
Comment (mjwitte) [reply tst/EnergyPlus/unit/UnitaryHybridAirConditioner.unit.cc:518]: Nope, nothing more to see here.

Comment (LipingWang) [reply tst/EnergyPlus/unit/UnitaryHybridAirConditioner.unit.cc:518]: It is related to the 'retaincase.' Previously, only object names in uppercase letters for ZoneHVAC:HybridUnitaryHVAC can be correctly recognized. With the change, the object name in mixed uppercase and lowercase letters can be correctly recognized using an existing unit test. It seems that additional lines are not showing here. Please refer to lines 579-603 for adding another object of ZoneHVAC:HybridUnitaryHVAC. In addition, a few checking function calls were added (lines 647-652) of this unit test.

Comment (Myoldmopar) [reply tst/EnergyPlus/unit/UnitaryHybridAirConditioner.unit.cc:518]: Yeah, it's fine.

**Comment by Myoldmopar** [comment]
I pulled up to latest develop just to make sure the unit test still built and it does fine.  And everything runs both serially and in parallel.  Good to go in.  Thanks @LipingWang

**Pull request merged**